### PR TITLE
Timers, Random Generators and Pickup Spawning

### PR DESCRIPTION
**Pull request by jormundur00**

### CODE REVIEW

File: src/Generator.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+I32 Generator::next_y()
+{
+	I32 next_element = y_coordinates.front();
+	y_coordinates.pop_front();
+	if (next_element == -1)
+	{
+		std::random_shuffle(y_coordinates.begin(), y_coordinates.end());
+		y_coordinates.push_back(-1);
```
Comment (gavrilovmiroslav) [inline src/Generator.cpp:45]: I think there's a bug here, but try it out with a small list of 3 elements. I think you're going to lose elements along the way, one every time...

File: src/Pickup.cpp [hunk 2]
```diff
@@ -25,24 +25,23 @@ void PickupSystem::on_tick()
...
 void PickupSystem::process_signal(PickupTimeoutSignal& signal)
 {
-	spawn()
-		.with<Pickup>("fastball" , 24.0f, 6)
-		.with<Sprite>(ecs::no_entity)
-		.with<SpriteAnimation>(Spritesheet::get_by_name("test/FastballTome"))
-		.with<Scale>(geometry::Vec2{0.8f, 0.8f})
-		.with<Position>(geometry::Vec2{ rand() % 600, rand() % 570 })
-		.with<Visibility>(true);
+	I32 x = MutAccessUnique<Generator>::access_unique().next_x();
```
Comment (gavrilovmiroslav) [inline src/Pickup.cpp]: Save the generator as a reference and use that several times

**Review by gavrilovmiroslav — APPROVED** [review]
Check for bug, but we're good to go. Let's fix everything in another branch where we don't have all the baggage :)

File: src/Generator.cpp (hunk 1 above)
Comment (jormundur00) [reply src/Generator.cpp:45]: Hmm. I tried printing out the deque every iteration and reducing the time between iterations to test it better, but I can't seem to replicate that bug. No elements are being lost on my end.

File: src/Pickup.cpp (hunk 2 above)
Comment (jormundur00) [reply src/Pickup.cpp]: Done.

**Pull request merged**